src/lib/Microsoft.Health.Extensions.Fhir.R4/IdentifierExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using Hl7.Fhir.Model;
+
+namespace Microsoft.Health.Extensions.Fhir
+{
+    public static class IdentifierExtensions
+    {
+        public static string ToSearchQueryParameter(this Identifier identifier)
+        {
+            EnsureArg.IsNotNull(identifier, nameof(identifier));
+
+            return $"identifier={identifier.System}|{identifier.Value}";
```

Comment:
>identifier.System}|{identifier.Value} [](http://example.com/codeflow?start=33&length=37)

I think it would be better to place this in the existing SearchExtensions class.  Also should use the already ToSearchToken.  The will properly prepare the search term for the identifier where the system| should be dropped if the system isn't specified on the identifier, i.e. 
        public static string ToSearchToken(this Hl7.Fhir.Model.Identifier identifier)

Reply:
Thanks for calling this out. I've removed the IdentifierExtensions class and leveraged ToSearchToken() instead

Reply:
Just wanted to confirm the ToSearchToken does everything you need.  I was originally just referring to using the ToSearchToken for composing the right hand side of the equal sign for identifier={identifier.System}|{identifier.Value}.

I believe with this change the ToSearchToken will only return the properly formatted identifier which will either be system|identifier or just identifer.  You may still need the identifer= portion.

I.E. the change would be to have your extension (moved to the common extension file) that did something like return $"identifier={identifier.ToSearchToken()}";

If the current code is working as expected no changes needed but would like to verify before completing the PR. Thanks!

Reply:
Thanks for pointing this out. I traced both ToSearchToken and ToSearchQueryParameter methods and their output is different as you suggested.

ToSearchQueryParameter
Input: { Value="android" System=null }
Output: identifier=|android

ToSearchToken
Input: { Value="android" System=null }
Output: android

My latest changes moved ToSearchQueryParameter to the common extension file and returns $"identifier={identifier.ToSearchToken()}" which leads to the following output for this test case

Latest changes
Input: { Value="android" System=null }
Output: identifier=android

Now the identifier is properly formatted for the search methods. Let me know if you have any further feedback
Thanks!